orangewidget/gui.py

```diff
@@ -619,16 +618,23 @@ def eventFilter(self, obj, event):
 
             pos = event.globalPos()
             posVal = pos.y() if self.verticalDirection else -pos.x()
-            diff = self.mouseStartPos - posVal
+            posValStart = self.mouseStartPos.y() if self.verticalDirection else -self.mouseStartPos.x()
+            diff = posValStart - posVal
+
             # these magic params are pretty arbitrary, ensure that it's still
             # possible to easily highlight the text if moving mouse slightly
             # up/down, with the default stepsize
-            valueOffset = int((diff / 25) ** 3) * self.stepSize
+            normalizedDiff = abs(diff) / 30
+            exponent = 1 + min(normalizedDiff / 10, 3)
+            valueOffset = int(normalizedDiff ** exponent) * self.stepSize
+            valueOffset = math.copysign(valueOffset, diff)
+
             self.setValue(self.preDragValue + valueOffset)
+
         elif event.type() == QEvent.MouseButtonRelease:
             # end click+drag
             # restore default cursor on release
-            if QApplication.overrideCursor() == cursor:
+            while QApplication.overrideCursor() is not None:
                 QApplication.restoreOverrideCursor()
 
             # restore click and hold behavior
```

Comment:
Now, that buttons are not connected to this event, could the `restore click and hold behavior` code section be removed?

Reply:
Also, the same for `# disable click and hold behavior` a bit higher.